src/api.h

```diff
@@ -0,0 +1,40 @@
+#ifndef MVM_API_H
+#define MVM_API_H
+
+#include <stddef.h>
+
+#ifndef MVM_TYPES_INCLUDED
```

Comment:
Any reason not to check this in types.h?

Reply:
Not really - I could put a standard inclusion guard

    #ifndef MVM_TYPES_H
    #define MVM_TYPES_H
    // ...
    #endif

around `types.h`. I didn't because all other checks were done in `api.h`, but I've no preference either way.

Reply:
Yes, I'd prefer it that way.